version: '3.7'

services:

  web:
    build:
      context: .
      args:
        django_secret_key: ${DJANGO_SECRET_KEY}
        django_debug: "False"
    container_name: web
    command: uwsgi uwsgi.ini
    restart: always
    volumes:
      -  /home/calum/pytho/static:/root/pytho/static
    ports:
      - "8000:8000"
    env_file:
      - .env
    depends_on:
      - db

  nginx:
    image: nginx:latest
    container_name: nginx
    restart: always
    volumes:
      -  /home/calum/pytho/pytho/uwsgi_params:/root/pytho/uwsgi_params
      -  /home/calum/pytho/static:/www/static
      - ./nginx:/etc/nginx/conf.d/nginx.conf
      -  /etc/letsencrypt:/etc/letsencrypt
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - web

  db:
    image: postgres:11
    container_name: db
    env_file:
      - .env
    volumes:
      - /home/calum/pytho/db/data:/var/lib/postgresql/data
      - /home/calum/pytho/db/shared:/root/shared

networks:
  default:
    external:
      name: pytho
